base/linalg/cholesky.jl

```diff
@@ -94,7 +94,8 @@ function chol!(x::Number, uplo)
     rxr = sqrt(rx)
     convert(promote_type(typeof(x), typeof(rxr)), rxr)
 end
-chol(x::Number, uplo::Symbol=:U) = chol!(x, uplo)
```

Comment:
Thanks for fixing this. I think it would be better just to remove the typing, i.e. `chol(x::Number, uplo) = chol!(x)`. Do you see any problems with that?